testing.html

```diff
@@ -4,20 +4,206 @@
     <title>Testing Phase</title>
     <script src="node_modules/jspsych/jspsych.js"></script>
     <script src="node_modules/jspsych/plugins/jspsych-instructions.js"></script>
+    <script src="node_modules/jspsych/plugins/jspsych-image-keyboard-response.js"></script>
+    <script src="node_modules/jspsych/plugins/jspsych-html-keyboard-response.js"></script>
+    <script src="utils.js"></script>
     <link href="predictiveaffect.css" rel="stylesheet" type="text/css"></link>
   </head>
   <body></body>
   <script>
 
+  // constants
   const timeline = [];
+  const imageStructLength = 3;
+
+  const exemplarsPerTripletType = 2;
+  const numTripletTypes = tripletTypes.length;
+  const numExemplars = exemplars.length;
+
+  const totalNumNegImages = negImages.length;
+  const totalNumNeuImages = neuImages.length;
+
+  // number of AAA triplets that have been tested on first A (no more than half of exemplarsPerTripletType)
+  var AAACount = 0;
+  // number of NNN triplets that have been tested on first N (no more than half of exemplarsPerTripletType)
+  var NNNCount = 0;
+  // number of images(one per exemplar) tested on same image type (no more than half of numExemplars)
+  var sameCatTested = 0;
+
+  // returns a random image 
+  const getRandImg = (curType, curTrip) => {
```

Comment:
In my testing I sometimes get two of the same images back to back with in the same triplet option. 

If this is invalid behavior I suggest we report this as a bug in zenhub and then merge what you have for now.